### PR TITLE
Add entropy minimization and refactor config usage

### PR DESCRIPTION
**Pull request by vuptran**

### CODE REVIEW

File: sesemi/models/heads/loss.py [hunk 1]
```diff
@@ -110,7 +118,49 @@ def forward(
...
+            input_data: The key used to get the unlabeled input data.
+            input_backbone: The key used to get the backbone for feature extraction.
+            logger: An optional PyTorch Lightning logger.
+        """
+        super().__init__(logger)
+        self.input_data = input_data
+        self.input_backbone = input_backbone
+
+    def build(self, backbones: Dict[str, Backbone], heads: Dict[str, Head], **kwargs):
+        self.fc_supervised = heads["supervised_head"]
```
Comment (tigist-d) [inline sesemi/models/heads/loss.py]: If shared heads are added as attributes they will be duplicated in the checkpoint on save. It is best to store only a single copy of the modules (either in the PyTorch `ModuleDict` or as an attribute to the `nn.Module`).

File: sesemi/models/heads/loss.py [hunk 2]
```diff
@@ -110,7 +118,49 @@ def forward(
...
+        data: Dict[str, Any],
+        backbones: Dict[str, Backbone],
+        heads: Dict[str, Head],
+        features: Dict[str, Any],
+        step: int,
+        **kwargs,
+    ) -> Tensor:
+        inputs_u, _ = data[self.input_data]
+        x_u = backbones[self.input_backbone](inputs_u)
+        output_u = self.fc_supervised(x_u)
```
Comment (tigist-d) [inline sesemi/models/heads/loss.py]: See the comment above about the use of the shared heads.

File: sesemi/models/heads/loss.py [hunk 3]
```diff
@@ -110,7 +118,49 @@ def forward(
...
+        output_u = F.softmax(output_u, dim=1)
+        loss_u = (-output_u * torch.log(output_u + 1e-5)).sum(1)
```
Comment (tigist-d) [inline sesemi/models/heads/loss.py:165]: You might consider the use of the `log_softmax` function here:
https://pytorch.org/docs/stable/generated/torch.nn.functional.log_softmax.html
